### PR TITLE
Fix/david/new link tab order

### PR DESCRIPTION
**Pull request by malmstein**
Task/Issue URL:  https://app.asana.com/0/414730916066338/1157894643192241

**Description**:
In the past, long tapping a link and opening it in a background tab, positioned the new tab immediately after the current tab. Now, when the new tab was added from any tab but the last one, it would not respect the proper order (See screencast for clarification)

**Solution**:
The code for adding a new tab was always using position = 0. The code for adding a tab in the background was properly adding 1 to the current position. For that reason, adding tabs in the background for the last tab was displayed properly, and not for the rest. This PR fixes that.

**Screencasts**:
| Before | After |
| ------ | ----- |
ezgif com-video-to-gif (1)| ezgif com-video-to-gif

**Steps to test this PR**:
On the current app in the Play Store
1. Open the app
2. Add a new tab (duckduckgo.com)
3. Add a second tab (wikipedia.com)
4. Switch to first tab
5. Long press on any link and select "Add tab in background"
6. New tab will be added as Tab in position 3 

Building this branch
1. Open the app
2. Add a new tab (duckduckgo.com)
3. Add a second tab (wikipedia.com)
4. Switch to first tab
5. Long press on any link and select "Add tab in background"
6. New tab will be added as Tab in position 2 (Between the current tab and wikipedia) 

---

### CODE REVIEW

File: app/src/main/java/com/duckduckgo/app/tabs/model/TabDataRepository.kt [hunk 1]
```diff
@@ -74,8 +74,14 @@ class TabDataRepository @Inject constructor(
                 return@scheduleDirect
             }
 
-            Timber.i("About to add a new tab, isDefaultTab: $isDefaultTab. $tabId")
-            val position = tabsDao.lastTab()?.position ?: 0
+            val lastTab = tabsDao.lastTab()
```
Comment (cmonfortep) [inline app/src/main/java/com/duckduckgo/app/tabs/model/TabDataRepository.kt]: Does it worth to include a test to cover this new logic?

File: app/src/main/java/com/duckduckgo/app/tabs/model/TabDataRepository.kt [hunk 2]
```diff
@@ -74,8 +74,14 @@ class TabDataRepository @Inject constructor(
...
+            val position = if (lastTab == null) {
+                0
+            }  else {
```
Comment (CDRussell) [inline app/src/main/java/com/duckduckgo/app/tabs/model/TabDataRepository.kt]: needs formatted

**Comment by malmstein** [comment]
@CDRussell @cmonfortep I've added a couple of tests to ensure the position is properly updated

**Comment by subsymbolic** [comment]
This is great, does it cover the crash recovery scenario mentioned in the task too? If not, you hack the refresh button to call the WebView crash url you can press refresh to test that scenario too:
```
fun refresh() {
  webView?.loadUrl("chrome://crash")
 }
```

**Comment by malmstein** [comment]
Hey @subsymbolic it does! The problem was adding the tab in the first place, so the subsequent tabs will be created properly.

**Pull request merged**